src/parser/content-evaluator-module.ts

```diff
@@ -113,22 +114,57 @@ export class ContentEvaluatorModule extends BaseModule {
     const commentsWithScore: GithubCommentScore[] = [...comments];
     const { commentsToEvaluate, prCommentsToEvaluate } = this._splitCommentsByPrompt(commentsWithScore);
 
-    const relevancesByAi = await this._evaluateComments(
-      specificationBody,
-      commentsToEvaluate,
-      allComments,
-      prCommentsToEvaluate
-    );
+    const relevancesByAi = await retry(
+      async () => {
+        const relevances = await this._evaluateComments(
+          specificationBody,
+          commentsToEvaluate,
+          allComments,
+          prCommentsToEvaluate
+        );
 
-    if (Object.keys(relevancesByAi).length !== commentsToEvaluate.length + prCommentsToEvaluate.length) {
-      throw this.context.logger.error("Relevance / Comment length mismatch!", {
-        expectedRelevances: commentsToEvaluate.length + prCommentsToEvaluate.length,
-        receivedRelevances: Object.keys(relevancesByAi).length,
-        relevancesByAi,
-        commentsToEvaluate,
-        prCommentsToEvaluate,
-      });
-    }
+        if (Object.keys(relevances).length !== commentsToEvaluate.length + prCommentsToEvaluate.length) {
+          throw this.context.logger.error("Relevance / Comment length mismatch!", {
```

Comment:
```suggestion
          throw this.context.logger.error("There was a mismatch between the relevance scores and amount of comments.", {
```

This errors should be written more naturally.